File: samples/catalog-app-shared/src/androidMain/kotlin/com/metacto/catalogapp/crash/FirebaseCrashlyticsImpl.kt
package com.metacto.catalogapp.crash

import com.google.firebase.crashlytics.FirebaseCrashlytics
import com.metacto.catalogapp.crash.FirebaseCrashlytics as FirebaseCrashlyticsInterface

class FirebaseCrashlyticsImpl : FirebaseCrashlyticsInterface {
    private val crashlytics by lazy { FirebaseCrashlytics.getInstance() }

    override fun setCustomKey(key: String, value: String) {
        crashlytics.setCustomKey(key, value)
    }

    override fun setUserId(userId: String) {
        crashlytics.setUserId(userId)
    }

    override fun log(message: String) {
        crashlytics.log(message)
    }

    override fun recordException(exception: Throwable) {
        crashlytics.recordException(exception)
    }
}

File: samples/catalog-ios-app/iosApp/Crash/FirebaseCrashlyticsImpl.swift
//
//  FirebaseCrashlyticsImpl.swift
//  iosApp
//
//  Created by Claude AI on 13/08/2025.
//  Copyright © 2025 orgName. All rights reserved.
//

import Foundation
import FirebaseCrashlytics
import appShared

class FirebaseCrashlyticsImpl: FirebaseCrashlytics {
    private let crashlytics = Crashlytics.crashlytics()
    
    func setCustomKey(key: String, value: String) {
        crashlytics.setCustomValue(value, forKey: key)
    }
    
    func setUserId(userId: String) {
        crashlytics.setUserID(userId)
    }
    
    func log(message: String) {
        crashlytics.log(message)
    }
    
    func recordException(exception: KotlinThrowable) {
        let nsError = NSError(
            domain: "com.metacto.catalogapp",
            code: 0,
            userInfo: [
                NSLocalizedDescriptionKey: exception.message ?? "Unknown error"
            ]
        )
        crashlytics.record(error: nsError)
    }
}